pkg/controllers/customcluster_controller.go

```diff
@@ -210,6 +233,26 @@ func (r *CustomClusterController) reconcile(ctx context.Context, customCluster *
 		return r.reconcileHandleProvisioning(ctx, customCluster, customMachine, cluster)
 	}
 
+	// Handle worker nodes scaling
+	if phase == v1alpha1.ProvisionedPhase || phase == v1alpha1.ScalingUpPhase || phase == v1alpha1.ScalingDownPhase {
```

Comment:
one special case, can both scale up and down at same time?

Reply:
Yes，current mechanism can handle this situation. 
For example, cm is (node1, node2) and customMachine is (node2, node3). 
To achieve the desired state, node3 needs to be added and node1 needs to be deleted. 

Here is the current processing flow: 
because both scaling up and scaling down conditions are met at the same time and the scaleup function is in front, scaleup will be executed first. 
After scaleup is completed, cm will become an intermediate state (node1, node2, node3), 
while customMachine still is (node2, node3) , which will automatically trigger scale down.

In general, when both scale up and down at same time, scale up will be automatically executed once, followed by an automatic execution of scale down.

Reply:
looks good to me up->down